gms/gossiper.cc

```diff
@@ -754,9 +754,30 @@ future<> gossiper::update_live_endpoints_version() {
     auto version = _live_endpoints_version + 1;
     return container().invoke_on_all([version] (gms::gossiper& g) {
         g._live_endpoints_version = version;
+        g._live_endpoints_version_condvar.broadcast();
     });
 }
 
+future<std::set<inet_address>> gossiper::get_live_members_synchronized(std::chrono::milliseconds timeout) {
+    auto live_members = gossiper::get_live_members();
+    auto version = _live_endpoints_version;
+    size_t nr_version = co_await container().map_reduce0([version, timeout] (gossiper& local_gossiper) -> future<size_t> {
+        try {
+            co_await local_gossiper._live_endpoints_version_condvar.when(timeout, [version, &local_gossiper] {
+                 return local_gossiper._live_endpoints_version >= version;
+            });
+        } catch (...) {
+            co_return 0;
+        }
+        co_return 1;
+    }, 0, std::plus<size_t>());
+    logger.debug("Nodes with gossiper version >= {}: {} / {}", version, nr_version, smp::count);
```

Comment:
Nodes?

Reply:
Fix 

> @alecco The endpoint crashes Scylla when you query it, it would be nice to try executing it at least once before sending a PR to review.
> 
> @gleb-cloudius @kostja I thought we wanted to fix the existing endpoint, I don't see much point introducing a new one. WDYT? @asias also WDYT?

Fix the existing one.

Reply:
Fix the existing on.